src/js/actions/type.js

```diff
@@ -661,25 +661,87 @@ define(function (require, exports) {
      * Initialize the list of installed fonts from Photoshop.
      *
      * @private
+     * @param {boolean=} force If true, re-initialize if necessary.
      * @return {Promise}
      */
-    var initFontList = function () {
+    var initFontList = function (force) {
         var fontStore = this.flux.store("font"),
             fontState = fontStore.getState(),
             initialized = fontState.initialized;
 
-        if (initialized) {
+        if (initialized && !force) {
             return Promise.resolve();
         }
 
         return descriptor.getProperty("application", "fontList")
             .bind(this)
-            .then(this.dispatch.bind(this, events.font.INIT_FONTS));
+            .then(this.dispatch.bind(this, events.font.INIT_FONTS))
+            .then(function () {
+                var document = this.flux.store("application").getCurrentDocument();
+                if (!document) {
+                    return;
+                }
+
+                var selected = document.layers.selected,
+                    typeLayers = selected.filter(function (layer) {
+                        return layer.isTextLayer();
+                    });
+
+                return this.transfer(layerActions.resetLayers, document, typeLayers);
```

Comment:
Yes, I would think you'd want to pass a third `true` argument to amend history.  As written, if you undo *back to* this state, the document won't reflect whatever changes were found by this resetLayers.